cmd/argoexec/commands/artifact/delete.go

```diff
@@ -82,13 +84,17 @@ func deleteArtifacts(labelSelector string, ctx context.Context, artifactGCTaskIn
 					return err
 				}
 
-				err = drv.Delete(&artifact)
-				if err != nil {
-					errString := err.Error()
-					artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: false, Error: &errString}
-				} else {
-					artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: true, Error: nil}
-				}
+				err = waitutil.Backoff(retry.DefaultRetry, func() (bool, error) {
+					err = drv.Delete(&artifact)
+					if err != nil {
+						errString := err.Error()
+						artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: false, Error: &errString}
+						return false, err
+					} else {
+						artResultNodeStatus.ArtifactResults[artifact.Name] = v1alpha1.ArtifactResult{Name: artifact.Name, Success: true, Error: nil}
+					}
+					return true, err
+				})
```

Comment:
This is a nitpick, but theoretically you don't need the else statement and you could just do that as part of the execution following the if statement.